Draft issue and PR templates.

real-yfprojects opened the pull request:
Includes `Bug Report`, `Bug Report Form`, `Feature Request` and a PR template.

This is still a draft since other part of the documentation have to be finished before the templates can be completed.

real-yfprojects committed:
Draft issue and PR templates.

Includes `Bug Report`, `Bug Report Form`, `Feature Request` and a PR template.

real-yfprojects commented:
Any suggestions for enhancing the templates?

akaihola commented on .github/ISSUE_TEMPLATE/bug_form.yaml:
```diff
@@ -0,0 +1,92 @@
+            I have searched the [darglint2 issue tracker](https://github.com/akaihola/darglint2/issues?q=is%3Aissue)
+            for open and closed issues to avoid duplicates.
+          required: true
+        - label: |
+            I have searched the [darglint issue tracker](https://github.com/terrencepreilly/darglint/issues?q=is%3Aissue)
+            for open and closed issues.
+          required: true
```
Why would we require people to search old Darglint issues? I think that's a bit of a burden and I can't immediately see the benefit.

real-yfprojects replied:
I was thinking that for the some issues the solution would already have been found in the old repository. Also there might be valuable information in old issues reporting the same bug. Including this in the templates, gives each reporter a bit more work to do but reduces the work we have in the end. However I am ok with dropping it if you think it doesn't provide that much benefit.

akaihola replied:
...or at least not requiring it?

real-yfprojects replied:
That's what I will go with for now.

akaihola commented on .github/ISSUE_TEMPLATE/bug_form.yaml:
```diff
@@ -0,0 +1,92 @@
+        Please create a minimal, reproducible example (MRE) that you have actually tested to work.
+        Please also use a minimal configuration and make sure that you don't overlook any config files in your working directory.
+        Here are some tips for providing a minimal example: https://stackoverflow.com/help/mcve
+        Please exclose the minimal example and the corresponding output.
```
Is "exclose" English? What does it mean?
```suggestion
        Please include the minimal example and the corresponding output.
```

real-yfprojects replied:
I meant `enclose`.

akaihola commented on .github/ISSUE_TEMPLATE/bug_form.yaml:
````diff
@@ -0,0 +1,92 @@
+      placeholder: |
+        I encountered a bug.
+
+        ```
+        minimal example (MRE)
+        ```
+
+        Steps to reproduce:
+        1. Save MRE to `somefile`
+        2. Run `some command` in bash
+        3. See error
````
```suggestion
        ```py
        # minimal example (MRE)
        ```

        Steps to reproduce:
        1. Save MRE to `somefile.py`
        2. Run `darglint2 <options>` in bash
        3. See error
```

akaihola commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,67 @@
+-->
+
+-   [ ] I have searched the [darglint2 issue tracker](https://github.com/akaihola/darglint2/issues?q=is%3Aissue) for open and closed issues to avoid duplicates.
+-   [ ] I have searched the [darglint issue tracker](https://github.com/terrencepreilly/darglint/issues?q=is%3Aissue) for open and closed issues.
```
Here, too, I question the purpose of asking to search old Darklint issues.
```suggestion
```

akaihola commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,67 @@
+Please create a minimal, reproducible example (MRE) that you have actually tested to work.
+Please also use a minimal configuration and make sure that you don't overlook any config files in your working directory.
+Here are some tips for providing a minimal example: https://stackoverflow.com/help/mcve
+Please exclose the minimal example and the corresponding output.
```
```suggestion
Please include the minimal example and the corresponding output.
```

akaihola commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,61 @@
+### Related Issue
+
+<!---
+This project only accepts pull requests related to open issues
```
What's the rationale behind this requirement? In some other projects, I've accepted PRs without corresponding issues provided that they had good descriptions.

real-yfprojects replied:
This is common practice. So I adopted it. The reasoning behind such policies is the following: An issue allows you to discuss a change before implementing it. In contrast a PR already contains changes which the contributor might have put lots of effort into. Rejecting the PR because it contains unwanted changes (e.g. out of the project's scope) can be hard for the maintainer because of all the work that went into the PR. Furthermore the contributor will feel very demotivated, possibly even angry that he did all the work for nothing.

akaihola replied:
On the other hand, I often find myself forking a project, poking around and trying out if I can easily fix a problem I've found, and then just submitting a PR if I succeeded. In such a case, I'd rather not be required to open an issue, describe the problem, open a PR, and describe the fix.

Anyway, it's not a big deal, I'm ok with this if it's common practice.

real-yfprojects replied:
Changed this, please have another look.

akaihola replied:
I like it.

akaihola commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,61 @@
+If you're unsure about any of these, don't hesitate to ask. We're here to help!
+-->
+
+-   [ ] I have read the [CONTRIBUTING]() guide.
+-   [ ] My code follows the [code style]() of this project.
```
Reminder: need to add links

real-yfprojects committed:
Update .github/ISSUE_TEMPLATE/bug_form.yaml

Co-authored-by: Antti Kaihola <13725+akaihola@users.noreply.github.com>

real-yfprojects committed:
Configure contact links for the issue form selector.

This configures that two contact links to the discussions and the docs will be displayed alongside the issue forms.

* .github/ISSUE_TEMPLATE/config.yml

real-yfprojects committed:
Fix typo in issue forms

real-yfprojects commented:
I configured so called contact links that will be listed below the issue forms.

real-yfprojects committed:
Adjust templates.

webknjaz commented:
> Any suggestions for enhancing the templates?

No need to duplicate Issue Forms with Issue Templates. Just use Issue Forms (yaml files).

akaihola commented:
> No need to duplicate Issue Forms with Issue Templates. Just use Issue Forms (yaml files).

@webknjaz does this mean that this PR can just be closed?

webknjaz commented:
Why? Just don't add templates, keep the forms.

akaihola commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1,8 @@
+contact_links:
+  - name: Documentation
+    url: https://akaihola.github.io/darglint2/
+    about: Documentation on installing, using and contributing to Vorta.
```
What is Vorta?
```suggestion
    about: Documentation on installing, using and contributing to Darglint2.
```

akaihola commented:
> Why? Just don't add templates, keep the forms.

Oops, indeed. So remove `bug_report.md` from the PR, right? @real-yfprojects could you do that?

webknjaz commented:
Yes, though it's probably useful to have several forms for different cases. Not just bugs. But it's up to you. I just wanted to point out that weird duplicate.

real-yfprojects committed:
Fix copy-paste error in config.yml

* .github/ISSUE_TEMPLATE/config.yml

Co-authored-by: Antti Kaihola <13725+akaihola@users.noreply.github.com>

real-yfprojects commented:
> > Any suggestions for enhancing the templates?
> 
> No need to duplicate Issue Forms with Issue Templates. Just use Issue Forms (yaml files).

Since issue forms are in beta still and I havened seen them used much yet, I decided to keep the option of the classic issue template which might be preferred by many people.

akaihola reviewed (changes requested):
I still see both
```
.github/ISSUE_TEMPLATE/bug_form.yaml
```
and
```
.github/ISSUE_TEMPLATE/bug_report.md
```

Aren't these the same thing both as a form and as a template, as pointed out by @webknjaz earlier?

real-yfprojects commented:
> @real-yfprojects could you do that?

I am no working on this PR or this repo in general. Anyone can pickup where I left of.

> Aren't these the same thing both as a form and as a template

Yes, they are.

webknjaz commented:
> Since issue forms are in beta still and I havened seen them used much yet, I decided to keep the option of the classic issue template which might be preferred by many people.

I've been using forms since before they've gone public, in a private beta. It's been a long time. They are quite stable. Here's an example: https://github.com/ansible/ansible/issues/new/choose.
